scr/main/java/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+
+    private static final String loginUserQuery = "SELECT * FROM " + UsersDBConstants.TABLE_USERS + " WHERE "
+            + UsersDBConstants.COLUMN_USERNAME + "=? AND " + UsersDBConstants.COLUMN_PASSWORD + "=? AND "
+            + UsersDBConstants.COLUMN_USERTYPE + "=?";
+
+    @Override
+    public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
+        Connection con = DBUtil.getConnection();
+        PreparedStatement ps;
+        User user = null;
+        try {
+            String userType = UserRole.SELLER.equals(role) ? "1" : "2";
+            ps = con.prepareStatement(loginUserQuery);
+            ps.setString(1, email);
+            ps.setString(2, password);
+            ps.setString(3, userType);
+            ResultSet rs = ps.executeQuery();
+            if (rs.next()) {
+                user = new User();
+                user.setFirstName(rs.getString("firstName"));
+                user.setLastName(rs.getString("lastName"));
+                user.setPhone(rs.getLong("phone"));
+                user.setEmailId(email);
+                user.setPassword(password);
+                session.setAttribute(role.toString(), user.getEmailId());
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return user;
+    }
+
+    @Override
+    public boolean isLoggedIn(UserRole role, HttpSession session) {
+        if (role == null)
+            role = UserRole.CUSTOMER;
+        return session.getAttribute(role.toString()) != null;
+    }
+
+    @Override
+    public boolean logout(HttpSession session) {
+        session.removeAttribute(UserRole.CUSTOMER.toString());
+        session.removeAttribute(UserRole.SELLER.toString());
+        session.invalidate();
+        return true;
+    }
+
+    @Override
+    public String register(UserRole role, User user) throws StoreException {
+        String responseMessage = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(registerUserQuery);
+            ps.setString(1, user.getEmailId());
+            ps.setString(2, user.getPassword());
+            ps.setString(3, user.getFirstName());
+            ps.setString(4, user.getLastName());
+            ps.setString(5, user.getAddress());
+            ps.setLong(6, user.getPhone());
+            ps.setString(7, user.getEmailId());
+            int userType = UserRole.SELLER.equals(role) ? 1 : 2;
+            ps.setInt(8, userType);
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                responseMessage = ResponseCode.SUCCESS.name();
+                ;
+            }
+        } catch (Exception e) {
+            responseMessage += " : " + e.getMessage();
+            if (responseMessage.contains("Duplicate"))
+                responseMessage = "User already registered with this email !!";
+            e.printStackTrace();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2418765539>#56fd03</a><span> - 10/17/2024, 07:44 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The current error handling in the register method may expose sensitive information. Consider implementing a more secure error handling mechanism.
</div>
<div id="fix">
<b>Fix:</b> Implement a generic error message for users and log the detailed error information securely for debugging purposes. Remove the direct exception message concatenation from the response.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -89,7 +89,9 @@
          } catch (Exception e) {
 -            responseMessage += " : " + e.getMessage();
 -            if (responseMessage.contains("Duplicate"))
 -                responseMessage = "User already registered with this email !!";
 +            responseMessage = "An error occurred during registration. Please try again later.";
 +            // Log the error securely for debugging
 +            logger.error("Registration error: ", e);
 +            // Consider removing or replacing e.printStackTrace() in production code
              e.printStackTrace();
          }
          return responseMessage;
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged